Add container build

jschlyter opened the pull request:
Add container (e.g., docker) build infrastructure. Should build, tag and publish container on release (`nsd:VERSION`) and on push to master (`nsd:latest`).

jschlyter committed:
Add container build infrastructure

jschlyter committed:
Run on master and container, after build-test has succeeded

jschlyter committed:
rename

jschlyter committed:
Build container on master only

jschlyter committed:
Mention volumes

jschlyter committed:
Merge remote-tracking branch 'upstream/master' into container

mozzieongit commented:
Hi, thanks for your contribution. The code/config look good.

However, providing container images for NSD is not something we want to maintain, currently. And we don't want to provide images now and then suddenly stop when something changes/breaks.

Your contribution would still be valuable to other users, so the whole `container` directory could be moved to the `contrib` directory and maintained by the community. The workflow would have to be removed.

jschlyter committed:
Move container infrastructure to contrib & remove workflow

jschlyter committed:
Adjust path to Dockerfile

jschlyter commented:
@mozzieongit I've removed the workflow and moved the `container` directory to `contrib`.

The solution to not breaking containers would be to always build them and fix them accordingly ;-)

mozzieongit commented:
> @mozzieongit I've removed the workflow and moved the `container` directory to `contrib`.

Thanks.

> The solution to not breaking containers would be to always build them and fix them accordingly ;-)

Certainly. It's just that _someone_ has to fix them.
